### PR TITLE
Refactor and tests

### LINKED ISSUE #252: [FEATURE]Create Officer and Incident Record View Tests
**Is your feature request related to a problem? Please describe.**
Officer and Incident record views could use unit tests.

**Describe the solution you'd like**
Close the issue by:
- Writing tests for officer and incident record views.

### PR DESCRIPTION
**Pull request by Logan-Dang**
Closes #252 

How to test refactor:
- <code>npm run dev</code>
- Navigate to /officer/0
- Confirm the officer page displays data
- Navigate to different officers using different ids (ex. /officer/12, /officer/1) [0..99] for now

How to test tests:
- <code>npm test</code>

### CODE REVIEW

File: frontend/pages/officer/[id].tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    )
+  }
+  else {
+    return (
+      <Layout>
+        <p>Loading...</p>
+      </Layout>
+    )
+  }
+
```
Comment (JMStudiosJoe) [inline frontend/pages/officer/[id].tsx:25]: we can remove commented out code but what about this file name? [id].tsx?

File: frontend/tests/snapshots/officer.test.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+import OfficerView from '../../compositions/officer-view'
+import { getOfficerFromMockData } from "../../helpers/mock-to-officer-type";
+
+describe('Officer Page', () => {
+    it('Renders the officer page correctly', () => {
+        const tree = render(<OfficerView {...getOfficerFromMockData(0)} />)
+        expect(tree).toMatchSnapshot()
+    })
+
+})
```
Comment (JMStudiosJoe) [inline frontend/tests/snapshots/officer.test.tsx:11]: need the EOF space

File: frontend/tests/snapshots/incident.test.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+import IncidentView from '../../compositions/incident-view'
+import getIncident from '../../helpers/incident'
+
+describe('Incident Page', () => {
+    it('Renders the incident page correctly', () => {
+        const tree = render(<IncidentView {...getIncident(0)} />)
+        expect(tree).toMatchSnapshot()
+    })
+
+})
```
Comment (JMStudiosJoe) [inline frontend/tests/snapshots/incident.test.tsx:11]: same here

File: frontend/pages/officer/[id].tsx (hunk 1 above)
Comment (JMStudiosJoe) [reply frontend/pages/officer/[id].tsx:25]: is it a NextJS thing?

**Comment by JMStudiosJoe** [comment]
Looks like just update the snapshots and the tests should pass.

**Comment by Logan-Dang** [comment]
JS Dates and Timezones

File: frontend/pages/officer/[id].tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    )
+  } else {
+    return (
+      <Layout>
+        <p>Loading...</p>
+      </Layout>
+    )
+  }
+
+  // try {
```
Comment (JMStudiosJoe) [inline frontend/pages/officer/[id].tsx:25]: can we remove the commented out code?

**Review by JMStudiosJoe — DISMISSED** [review]
just a minor comment but looks great

**Review by JMStudiosJoe — APPROVED** [review]

**Pull request merged**